src/fairseq2/models/llama/factory.py

```diff
@@ -217,8 +240,14 @@ def build_attention(
         sdpa = create_default_sdpa(attn_dropout_p=self._config.dropout_p)
 
         if self._pos_encoder is None:
-            if self._config.use_scaled_rope:
-                freqs_init_fn = self._init_scaled_freqs
+            if self._config.rope_scaling is not None:
+                freqs_init_fn = functools.partial(
```

Comment:
Nit: how about `_init_scaled_freqs(pos_encoder: RotaryEncoder, scaling: RopeScaling)` to make the signature shorter?